src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.Android.cs

```diff
@@ -65,8 +64,17 @@ public static DateTimeOffset Now
                         // setting the boolean flag to false immediately after should
                         // prevent two calls to DateTimeOffset.Now in quick succession
                         // from both reaching here.
-                        s_loadAndroidTZData.Start();
-                        s_startNewBackgroundThread = false;
+                        //
+                        // In the event multiple threads hit Start at the same time,
+                        // swallow the exception and move on.
```

Comment:
This isn't the right way to address the problem.  The right way is avoiding the situation in the first place.

Reply:
Do you have a suggestion for an alternative approach?  During the review in main, we originally had the call to Start occur in the outer lock.  We changed that because there was a rare potential for a deadlock.

Reply:
Maybe something like this is what Stephen had in mind:

```csharp
partial struct DateTimeOffset {
  internal enum BackgroundThreadState {
    NotStarted,
    Running,
    Finished,
  }

  private static BackgroundThreadState s_backgroundThread = BackgroundThreadState.NotStarted;
  private static object s_localUtcOffsetLock = new object();
  private static  boolean s_androidTZDataLoaded;

  public static DateTimeOffset Now
  {
    get {
      if (s_androidTZDataLoaded)
         return ...;

     Thread? backgroundThread = null;
     lock (s_localUtcOffsetLock)
     {
       if (s_androidTZDataLoaded)
         return ...;

        if (s_backgroundThread == BackgroundThreadState.NotStarted) {
           backgroundThread = new Thread(BackgroundThreadBody, { IsBackground = true});
           s_backgroundThread = BackgroundThreadState.Running;
       }
     }
    // backgroundThread local var is assigned under the lock only in the thread that that will start the background thread.
    if (backgroundThread != null)
      backgroundThread.Start();

    // otherwise return the approximate result
    ...
}

private static void BackgoundThreadBody() {
  Thread.Sleep (1000);
  // do the loading
  ...
  lock (s_localUtcOffsetLock) {
    s_backgroundThread = BackgroundThreadState.Finished;
    s_androidTZDataLoaded = true;
  }
}
```

This can probably be further simplified by dropping the `Finished` state since noone checks for it (ie even after the thread is done just leave `s_backgroundThread == Running`) but the above is probably easier to understand.

The key points:
1. `backgroundThread` is a local var not a static.
2. `backgroundThread` is assigned inside the lock by only one thread; it's started outside the lock
3. `s_backgroundThread` goes from `NotStarted` to `Running` to `Finished`.

Reply:
> Maybe something like this is what Stephen had in mind

Yup. Except doing your "This can probably be further simplified" and just having a `private static bool s_backgroundThreadStarted;`. You also don't need the `backgroundThread = null; // no longer needed`, so that `if` block just becomes `backgroundThread?.Start();`. Etc.

Reply:
I was imagining something like this:
```C#
partial struct DateTimeOffset
{
    private static int s_androidTZDataLoaded = -1; // 0 == in process of being loaded, 1 == loaded

    public static DateTimeOffset Now
    {
        get
        {
            if (s_androidTZDataLoaded == 1)
                return ...; // do the real thing knowing the data has been loaded

            if (Interlocked.CompareExchange(ref s_androidTZDataLoaded, 0, -1) != -1)
                return ...; // do the approximate thing, as the data is still being loaded

           new Thread(() =>
            {
                Thread.Sleep (1000);
                try { ... // load the data } catch
                finally { Volatile.Write(ref s_androidTZDataLoaded, 1); }
            }) { IsBackground = true }.Start();
        }
    }
}
```

Reply:
> We're not blocking a thread for a second, nor spinning up and tearing down a thread outside of the pool.

I think the threadpool hasn't been initialized at this point in android startup. that's why we're spinning up a dedicated thread instead.

Reply:
Unfortunate, but ok, in that case instead of:
```C#
           Task.Delay(1000).ContinueWith(delegate
           {
                try { ... /* load the data */ }
                finally { Volatile.Write(ref s_androidTZDataLoaded, 1); }
           });
```
it'd be:
```C#
           new Thread(() =>
           {
                try { ... /* load the data */ }
                finally { Volatile.Write(ref s_androidTZDataLoaded, 1); }
           }) { IsBackground = true }.Start();
```